src/acuvity/response/evaluator.py

```diff
@@ -0,0 +1,76 @@
+from typing import Optional
+
+from ..guard.config import Guard
+from ..models.scanresponse import Scanresponse
+from ..utils.logger import get_default_logger
+from .parser import ResponseParser
+from .result import GuardMatch, ResponseMatch
+
+logger = get_default_logger()
+
+class ResponseEvaluator:
+    """
+    Handles pure check evaluation without considering actions.
+    This evaluator determines if conditions are met based on thresholds.
+    """
+
+    def __init__(self):
+        self._parser = ResponseParser()  # Use the existing ResponseParser
+        self._response = None
+
+    def evaluate(
+        self,
+        response: Scanresponse,
+        guard: Guard,
+        match_name: Optional[str] = None
+    ) -> GuardMatch:
+        """
+        Evaluates a check condition using a Threshold object.
+
+        Args:
+            response: The response data to check
+            path: Path to the value in the response
+            threshold: Threshold object containing value and operator
+
+        Returns:
+            GuardMatch with PASS if condition met, FAIL if not met
+        """
+        try:
+            if not response.extractions:
+                raise ValueError("No extractions found in the response")
+            extraction = response.extractions[0]
```

Comment:
uhh... so what happens when there were multiple inputs?

Reply:
is the msg to extraction mapping, 1-1, like 1st msg links to 1st extraction.
the backend handlescan makes sure of it, right ?